[rpc] add num_pending_users to debug info

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31381 [rpc][profiler] add support for profiling rpc.(a)sync, rpc.remote calls.
* **#31539 [rpc] add num_pending_users to debug info**

Adding this metric primarily because it is needed to unblock unit
tests for https://github.com/pytorch/pytorch/pull/31381. It also may be useful
to look at this metric to see the number of pending RRef forks that currently
exist.

Differential Revision: [D19204158](https://our.internmc.facebook.com/intern/diff/D19204158/)

rohan-varma committed:
[rpc] add num_pending_users to debug info

Adding this metric primarily because it is needed to unblock unit
tests for https://github.com/pytorch/pytorch/pull/31381. It also may be useful
to look at this metric to see the number of pending RRef forks that currently
exist.

Differential Revision: [D19204158](https://our.internmc.facebook.com/intern/diff/D19204158/)

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 35f4460a:


* **1/1** broken upstream at merge base 3a19980b (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=3a19980b78b6638235edef367784ecc3dc37e364&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
  <details>
  <summary>
  You may want to rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
  ```
  git fetch origin viable/strict
  git rebase --onto viable/strict $(git merge-base origin/master HEAD)
  ```
  If your commit is older than `viable/strict`:
  ```
  git fetch origin viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  
* **0/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=35f4460a8c61267e28aa0ecc992ab390adcecc0f).

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably caused by upstream breakages:
* [pytorch_linux_xenial_py3_clang5_asan_test](https://dr.pytorch.org/build-details.html?build_id=78526936)



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

mrshenli reviewed (approved):
Would I be correct if i assume there is no reliable way to test non-zero `num_pending_users` value?

rohan-varma commented:
> Would I be correct if i assume there is no reliable way to test non-zero `num_pending_users` value?

Yeah, I'm not so sure how to test this. We'd need a way to be able to do the check in between having sent the RRef fork request, but before having processed the response from the owner worker. I'm not sure if there's a way to guarantee that the owner worker hasn't sent the confirmation yet, since all of the processing is done in background threads.

rohan-varma committed:
Update on "[rpc] add num_pending_users to debug info"


Adding this metric primarily because it is needed to unblock unit
tests for https://github.com/pytorch/pytorch/pull/31381. It also may be useful
to look at this metric to see the number of pending RRef forks that currently
exist.

Differential Revision: [D19204158](https://our.internmc.facebook.com/intern/diff/D19204158/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] add num_pending_users to debug info"


Adding this metric primarily because it is needed to unblock unit
tests for https://github.com/pytorch/pytorch/pull/31381. It also may be useful
to look at this metric to see the number of pending RRef forks that currently
exist.

Differential Revision: [D19204158](https://our.internmc.facebook.com/intern/diff/D19204158/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] add num_pending_users to debug info"


Adding this metric primarily because it is needed to unblock unit
tests for https://github.com/pytorch/pytorch/pull/31381. It also may be useful
to look at this metric to see the number of pending RRef forks that currently
exist.

Differential Revision: [D19204158](https://our.internmc.facebook.com/intern/diff/D19204158/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] add num_pending_users to debug info"


Adding this metric primarily because it is needed to unblock unit
tests for https://github.com/pytorch/pytorch/pull/31381. It also may be useful
to look at this metric to see the number of pending RRef forks that currently
exist.

Differential Revision: [D19204158](https://our.internmc.facebook.com/intern/diff/D19204158/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] add num_pending_users to debug info"


Adding this metric primarily because it is needed to unblock unit
tests for https://github.com/pytorch/pytorch/pull/31381. It also may be useful
to look at this metric to see the number of pending RRef forks that currently
exist.

Differential Revision: [D19204158](https://our.internmc.facebook.com/intern/diff/D19204158/)

[ghstack-poisoned]
